venafi/resource_venafi_certificate.go

```diff
@@ -361,9 +362,63 @@ func resourceVenafiCertificateUpdate(_ context.Context, d *schema.ResourceData,
 	return nil
 }
 
-func resourceVenafiCertificateDelete(_ context.Context, d *schema.ResourceData, _ interface{}) diag.Diagnostics {
-	// We don't support deletion for created certificates from TPP, so we just remove it from state
+func resourceVenafiCertificateDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) (diags diag.Diagnostics) {
+
+	//casting meta interface to the expected *venafiProviderConfig
+	provConf, ok := meta.(*venafiProviderConfig)
+	if !ok {
+		castError := errors.New(messageVenafiProviderConfigCastingFailed)
+		tflog.Error(ctx, castError.Error())
+		return diag.FromErr(castError)
+	}
+	vCertCfg := provConf.vCertCfg
+
+	//VCert connector fake has not implemented the RetireCertificate functionality
+	if provConf.skipRetirement || vCertCfg.ConnectorType == endpoint.ConnectorTypeFake {
+		// removing it from state
+		d.SetId("")
+		return nil
+	}
+
+	cl, err := getConnection(ctx, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	certID := d.Id()
+	parameters := strings.Split(certID, ",")
+
+	var pickupID string
+
+	// When retrieving the certID we have to watch for two cases: when certificate is imported and when is not
+	// For both cases we get the pickupID from the first parameter and the key password from state
+	pickupID = parameters[0]
+
+	// But we need to make extra verifications if state was tainted by a third party.
+	// We ignore the case when parameters length is equal to 1, since that's standard accepted case when certificate is not imported.
+	if len(parameters) < 1 {
+		return buildStantardDiagError(fmt.Sprintf("%s: certID was not found from terraform state", terraformStateTainted))
```

Comment:
message should be: 
`certificate ID not found in terraform state`

Reply:
This is a copy, not a variable reference, we should take this opportunity to have this message written properly. Thus,  the only ones needing fixing would be the ones in `resourceVenafiCertificateRead()`. Otherwise we are introducing another syntactically incorrect message

Reply:
done